cmd/operator-sdk/main.go

```diff
@@ -81,3 +84,70 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func checkDepManagerForCmd(cmd *cobra.Command) (err error) {
+	// Do not perform this check if wd is not in the project root,
+	// as some sub-commands might not require project root.
+	if err := projutil.CheckProjectRoot(); err != nil {
+		return nil
+	}
+
+	var dm projutil.DepManagerType
+	switch n := cmd.Name(); n {
+	case "new", "migrate":
```

Comment:
I need to test this out but when you do `operator-sdk new ...` to create a new project, you're not in a project's root directory (aka no `build/Dockerfile` in the wd).
Which means `CheckProjectRoot()` will return an error and we return from Line 92 above.

And if you've already created a project and you run `operator-sdk new` inside a project's root then the command should fail. Which means there's no need to check for the type of dep manager at that point.

So maybe I'm missing something but it seems to me we will never reach this case for `new` when we actually need it i.e creating a new project when not inside a root directory.

Reply:
I'm going to move `new` and `migrate` dep checks to their respective `cmd` packages. There are stack overflow issues if the expected flags are not set properly, and they complicate the logic in `main.go`. Once moved, we can skip checking for project root.